packages/components/bin/build-metadata.mts

```diff
@@ -0,0 +1,76 @@
+import { mkdir, writeFile } from 'node:fs/promises';
+import { join } from 'node:path';
+import { parse } from 'react-docgen-typescript';
+
+type ComponentMetadata = {
+	props: Record<
+		string,
+		{
+			description: string;
+			defaultValue?: { value: any };
+			required: boolean;
+			type: { name: string };
+		}
+	>;
+};
+
+const DOCUMENTED_COMPONENTS = [ 'breadcrumbs', 'summary-button' ];
+
+const files = DOCUMENTED_COMPONENTS.map( ( component ) => `${ component }/index.tsx` );
+
+const pick = < O extends Record< string, any >, K extends Array< keyof O > >(
+	obj: O,
+	keys: K
+): Pick< O, K[ number ] > =>
+	Object.fromEntries( Object.entries( obj ).filter( ( [ key ] ) => keys.includes( key ) ) ) as Pick<
+		O,
+		K[ number ]
+	>;
+
+const omit = < O extends Record< string, any >, K extends Array< keyof O > >(
+	obj: O,
+	keys: K
+): Omit< O, K[ number ] > =>
+	Object.fromEntries(
+		Object.entries( obj ).filter( ( [ key ] ) => ! keys.includes( key ) )
+	) as Omit< O, K[ number ] >;
+
+const mapValues = < V, MV >(
+	obj: Record< string, V >,
+	fn: ( value: V ) => MV
+): Record< string, MV > =>
+	Object.fromEntries( Object.entries( obj ).map( ( [ key, value ] ) => [ key, fn( value ) ] ) );
```

Comment:
This is a good example of where rolling our own utilities is not so fun 🙃 (Re: https://github.com/Automattic/wp-calypso/pull/103048#discussion_r2071495119)